deployment/ccip/changeset/v1_5/test_helpers.go

```diff
@@ -0,0 +1,529 @@
+	"github.com/ethereum/go-ethereum/common"
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+	"github.com/smartcontractkit/chainlink-common/pkg/config"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccip"
+	jobv1 "github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+	"github.com/smartcontractkit/chainlink-testing-framework/lib/utils/testcontext"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/confighelper"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset"
+	commonchangeset "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	commontypes "github.com/smartcontractkit/chainlink/deployment/common/types"
+	"github.com/smartcontractkit/chainlink/deployment/environment/memory"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/commit_store"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/evm_2_evm_offramp"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/evm_2_evm_onramp"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/price_registry_1_2_0"
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/ccip/testhelpers"
+	ocr2validate "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/validate"
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocrbootstrap"
+)
+
+type LegacyJobclient struct {
+	*memory.JobClient
+}
+
+// ProposeJob is an overridden implementation of the jobclient.ProposeJob method which allows offchainreporting2 job type
+func (j *LegacyJobclient) ProposeJob(ctx context.Context, in *jobv1.ProposeJobRequest, opts ...grpc.CallOption) (*jobv1.ProposeJobResponse, error) {
+	n := j.Nodes[in.NodeId]
+	// TODO: Use FMS
+	jb, err := ocr2validate.ValidatedOracleSpecToml(
+		ctx,
+		n.App.GetConfig().OCR2(),
+		n.App.GetConfig().Insecure(),
+		in.Spec,
+		nil, // not required for validation
+	)
+	if err != nil {
+		jb, err = ocrbootstrap.ValidatedBootstrapSpecToml(in.Spec)
+		if err != nil {
+			return nil, fmt.Errorf("failed to validate job spec only bootstrap and offchainreporting2 are supported : %w", err)
+		}
+	}
+	err = n.App.AddJobV2(ctx, &jb)
+	if err != nil {
+		return nil, err
+	}
+	return &jobv1.ProposeJobResponse{Proposal: &jobv1.Proposal{
+		Id: "",
+		// Auto approve for now
+		Status:             jobv1.ProposalStatus_PROPOSAL_STATUS_APPROVED,
+		DeliveryStatus:     jobv1.ProposalDeliveryStatus_PROPOSAL_DELIVERY_STATUS_DELIVERED,
+		Spec:               in.Spec,
+		JobId:              jb.ExternalJobID.String(),
+		CreatedAt:          nil,
+		UpdatedAt:          nil,
+		AckedAt:            nil,
+		ResponseReceivedAt: nil,
+	}}, nil
+}
+
+type MemoryEnvironment struct {
```

Comment:
I think the overriding will be confusing for the migration test involving 1.5 and 1.6 - can we not have a constructor/flag/config in memory.NewEnvironment which ensures there's a 1337 chain available?